presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -269,8 +269,8 @@ public HiveTableHandle getTableHandle(ConnectorSession session, SchemaTableName
             return null;
         }
 
-        if (isPartitionsSystemTable(tableName)) {
-            // We must not allow $partitions table due to how permissions are checked in PartitionsAwareAccessControl.checkCanSelectFromTable()
+        if (getSourceTableNameFromSystemTable(tableName).isPresent()) {
+            // We must not allow system table due to how permissions are checked in PartitionsAwareAccessControl.checkCanSelectFromTable()
```

Comment:
Let's rename `PartitionsAwareAccessControl` to `SystemTableAwareAccessControl` since it's not handling more than just partitions

Reply:
done